src/api/webhook.py

```
@@ -32,28 +33,47 @@ def verify_signature(body: bytes, signature: str) -> bool:
+                    logger.info(f"Processing text message from {user_id}: {message_text}")
+                    
+                    # Process message and send reply
+                    await handle_text_message_direct(user_id, message_text, reply_token)
         
         return {"status": "success"}
         
-    except InvalidSignatureError:
-        raise HTTPException(status_code=400, detail="Invalid signature")
+    except json.JSONDecodeError as e:
+        logger.error(f"Invalid JSON in webhook body: {e}")
+        raise HTTPException(status_code=400, detail="Invalid JSON")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)